src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -27,21 +27,21 @@ public TorpedoStore(int numberOfTorpedos){
       }
     }
   }
-
+  Random generator = new Random();
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+    
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```

Comment:
This is correct syntax for this.torpedoCount =this.torpedoCount - numberOfTorpedos;